src/seq/nucleotide.jl

```diff
@@ -97,27 +97,27 @@ end
 
 Test if nucleotide is surely a purine.
 """
-function ispurine(nt::Nucleotide)
+@inline function ispurine(nt::Nucleotide)
     bits = reinterpret(UInt8, nt)
-    return bits != 0 && (bits & 0b1010) == 0
+    return (bits != 0) & ((bits & 0b1010) == 0)
```

Comment:
I've just checked on my home laptop and I'm seeing simd instructions and equal performance with the short-circuit operations now. The only other explanation is I've recently updated a load of software this week, including R and julia, and perhaps I started this work on one version and moved to another and that explains the difference. Either way, we get simd without changing those operations now! I'll revert.